api/kubernetes/yaml.go

```diff
@@ -55,10 +103,29 @@ func AddAppLabels(manifestYaml []byte, appLabels KubeAppLabels) ([]byte, error)
 		docs = append(docs, out.Bytes())
 	}
 
-	return bytes.Join(docs, []byte("---\n")), nil
+	return docs, nil
+}
+
+// GetNamespace returns the namespace of a kubernetes resource from its metadata
+// It returns an empty string if namespace is not found in the resource
+func GetNamespace(manifestYaml []byte) (string, error) {
+	yamlDecoder := yaml.NewDecoder(bytes.NewReader(manifestYaml))
+	m := make(map[string]interface{})
+	err := yamlDecoder.Decode(&m)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to unmarshal yaml manifest when obtaining namespace")
+	}
+
+	if _, ok := m["metadata"]; ok {
+		if namespace, ok := m["metadata"].(map[string]interface{})["namespace"]; ok {
+			return namespace.(string), nil
+		}
+	}
+
+	return "", nil
```

Comment:
Just my opinion but I think it's generally better and easier to read to have normal success flow out the bottom and early exit/return inside the if blocks.  Not only does it make it easier to see where the success flow should exit from, but it reduces indenting of if blocks.

This is a trivial example though so not too concerned.   I would think something along the lines of

```
metadata, err := m["metadata"]
if err != nil {
  "", errors.Wrap(err, "missing metadata")
}

namespace, err := metadata.(map[string]interface{})["namespace"]
if err != nil ...
...
```

Reply:
Additionally we dont want to return error if the namespace is not found (as it is an optional field on manifest).